src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -8,60 +9,95 @@ namespace NServiceBus
+            {
+                await MoveMessageToErrorQueue(context, fork, message, failureInfo.Exception).ConfigureAwait(false);
+
+                return;
+            }
+
             try
             {
                 await next().ConfigureAwait(false);
             }
-            catch (Exception exception)
+            catch (Exception ex)
             {
-                try
+                if (RunningWithTransactions)
                 {
-                    var message = context.Message;
+                    failureInfoStorage.MarkForMovingToErrorQueue(message.MessageId, ExceptionDispatchInfo.Capture(ex));
 
-                    Logger.Error($"Moving message '{message.MessageId}' to the error queue because processing failed due to an exception:", exception);
+                    context.AbortReceiveOperation();
+                }
+                else
+                {
+                    await MoveMessageToErrorQueue(context, fork, message, ex).ConfigureAwait(false);
+                }
+            }
+        }
 
-                    message.RevertToOriginalBodyIfNeeded();
+        async Task MoveMessageToErrorQueue(ITransportReceiveContext context, Func<IFaultContext, Task> fork, IncomingMessage message, Exception exception)
+        {
+            try
+            {
+                Logger.Error($"Moving message '{message.MessageId}' to the error queue because processing failed due to an exception:", exception);
 
-                    message.SetExceptionHeaders(exception, localAddress);
+                message.RevertToOriginalBodyIfNeeded();
 
-                    message.Headers.Remove(Headers.Retries);
+                message.SetExceptionHeaders(exception, localAddress);
 
-                    var outgoingMessage = new OutgoingMessage(message.MessageId, message.Headers, message.Body);
-                    var faultContext = this.CreateFaultContext(context, outgoingMessage, errorQueueAddress, exception);
+                message.Headers.Remove(Headers.Retries);
+                message.Headers.Remove(Headers.FLRetries);
 
-                    await fork(faultContext).ConfigureAwait(false);
+                var outgoingMessage = new OutgoingMessage(message.MessageId, message.Headers, message.Body);
+                var faultContext = this.CreateFaultContext(context, outgoingMessage, errorQueueAddress, exception);
 
-                    await context.RaiseNotification(new MessageFaulted(message, exception)).ConfigureAwait(false);
-                }
-                catch (Exception ex)
-                {
-                    criticalError.Raise("Failed to forward message to error queue", ex);
-                    throw;
-                }
+                await fork(faultContext).ConfigureAwait(false);
+
+                await context.RaiseNotification(new MessageFaulted(message, exception)).ConfigureAwait(false);
+
+                failureInfoStorage.ClearFailureInfoForMessage(message.MessageId);
```

Comment:
That's correct. If the user tries to manually reply the message and it's still in in-memory failure storage it will be immediately moved to `error` queue.